apps/backend/src/libs/modules/token/base-token.module.ts

```diff
@@ -0,0 +1,38 @@
+import {
+	type JWTPayload,
+	jwtVerify,
+	type JWTVerifyResult,
+	SignJWT,
+} from "jose";
+
+type Constructor = {
+	algorithm: string;
+	expirationTime: string;
+	secret: Uint8Array;
+};
+
+class BaseToken<T extends JWTPayload> {
+	private algorithm: string;
+	private expirationTime: string;
+	private secret: Uint8Array;
+
+	constructor({ algorithm, expirationTime, secret }: Constructor) {
+		this.secret = secret;
+		this.algorithm = algorithm;
+		this.expirationTime = expirationTime;
+	}
+
+	public async sign(payload: T): Promise<string> {
+		return await new SignJWT(payload)
+			.setProtectedHeader({ alg: this.algorithm })
+			.setIssuedAt()
+			.setExpirationTime(this.expirationTime)
+			.sign(this.secret);
+	}
```

Comment:
Seems like no async await needed here. The same for verify method
```suggestion
	public sign(payload: T): Promise<string> {
		return new SignJWT(payload)
			.setProtectedHeader({ alg: this.algorithm })
			.setIssuedAt()
			.setExpirationTime(this.expirationTime)
			.sign(this.secret);
	}
```